### ISSUE TITLE
zero should be mutable in `invperm`

### ISSUE THREAD

**Issue by jishnub** [JuliaLang/julia#47847]
https://github.com/JuliaLang/julia/blob/6bfc6ac7187f04cc2554425ecc8a2fa42b4f1af8/base/combinatorics.jl#L279-L285

On line 279, this should be `fill!(similar(a), zero(eltype(a)))`, as `zero(a)` may not be mutable (e.g. for `StaticArray`s).

**Comment by shogohida** [comment]
Hi @jishnub, can I work on this?

**Comment by jishnub** [comment]
Yes, of course, feel free to make a PR. Please include a test (perhaps we might need to create a custom array type for which zeros returns an immutable, I'm not sure if there's such an array type in Base)

**Comment by shogohida** [comment]
Thanks! I'm new to Julia so I might need your help. I'll let you know when I make a PR!
